### PR TITLE
r/azurerm_windows_virtual_machine and r/azurerm_windows_virtual_machine_scale_set - secure_boot_enabled, security_type, vtpm_enabled

### LINKED ISSUE #13039: Support for full SecurityProfile for azurerm_windows_virtual_machine
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Can you please add support for adding a Security Profile block when deploying a Windows Virtual Machine. One of the Security Profile properties has already been implemented outside of a Security Profile block, however Azure has now added other attributes as part of the security profile to support Trusted Launch of Virtual Machines.  

### New or Affected Resource(s)

* azurerm_windows_virtual_machine
* azurerm_windows_virtual_machine_scale_set

### Potential Terraform Configuration

```hcl
resource "azurerm_windows_virtual_machine" "example" {
  name                = "example-machine"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  size                = "Standard_F2"
  admin_username      = "adminuser"
  admin_password      = "P@$$w0rd1234!"
  network_interface_ids = [
    azurerm_network_interface.example.id,
  ]

  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "Standard_LRS"
  }

  security_profile {
    encryptionAtHost  = true
      securityType = 'TrustedLaunch'
      uefiSettings {
        secureBootEnabled = true
        vTpmEnabled = true
      }
  }

  source_image_reference {
    publisher = "MicrosoftWindowsServer"
    offer     = "WindowsServer"
    sku       = "2016-Datacenter"
    version   = "latest"
  }
}
```

### References

* https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#securityprofile

### PR DESCRIPTION
**Pull request by amccullough84**
Adds secure_boot_enabled and vtpm_enabled properties for windows_virtual_machine and windows_virtual_machine_scale_set.

Fixes #13039

### CODE REVIEW

File: internal/services/compute/windows_virtual_machine_scale_set_other_resource_test.go [hunk 1]
```diff
@@ -677,6 +677,93 @@ func TestAccWindowsVirtualMachineScaleSet_otherEncryptionAtHostEnabledWithCMK(t
...
+		data.ImportStep("admin_password"),
+		{
+			Config: r.otherVTpmEnabled(data, true),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("admin_password"),
+	})
+}
```
[the remark this answers is not in the document]
Comment (amccullough84) [reply internal/services/compute/windows_virtual_machine_scale_set_other_resource_test.go]: Removed update test

File: internal/services/compute/windows_virtual_machine_scale_set_resource.go [hunk 2]
```diff
@@ -1105,10 +1152,27 @@ func resourceWindowsVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta i
 		d.Set("extensions_time_budget", extensionsTimeBudget)
 
 		encryptionAtHostEnabled := false
-		if profile.SecurityProfile != nil && profile.SecurityProfile.EncryptionAtHost != nil {
-			encryptionAtHostEnabled = *profile.SecurityProfile.EncryptionAtHost
+		securityType := ""
```
Comment (tombuildsstuff) [inline internal/services/compute/windows_virtual_machine_scale_set_resource.go:1155]: we're not using this value anymore, so we can remove this:

```suggestion
```

File: internal/services/compute/windows_virtual_machine_resource.go [hunk 3]
```diff
@@ -725,10 +772,27 @@ func resourceWindowsVirtualMachineRead(d *pluginsdk.ResourceData, meta interface
 	}
 
 	encryptionAtHostEnabled := false
-	if props.SecurityProfile != nil && props.SecurityProfile.EncryptionAtHost != nil {
-		encryptionAtHostEnabled = *props.SecurityProfile.EncryptionAtHost
+	securityType := ""
```
Comment (tombuildsstuff) [inline internal/services/compute/windows_virtual_machine_resource.go:775]: this isn't used, so this'll get picked up as a linter error:

```suggestion
```

**Review by tombuildsstuff — APPROVED** [review]
hey @amccullough84 

Thanks for pushing those changes - apologies for the delayed re-review here, I've spotted a couple of minor things which'll mean this doesn't build/test correctly - but if we can remove the unused lines (which I hope you don't mind but I'm going to push a commit to fix) then this should otherwise be good to go 👍 

Thanks!

**Comment by tombuildsstuff** [comment]
Tests look good 👍

**Pull request merged**